src/client/common/utils/async.ts

```diff
@@ -110,3 +110,94 @@ export function createDeferredFromPromise<T>(promise: Promise<T>): Deferred<T> {
     promise.then(deferred.resolve.bind(deferred)).catch(deferred.reject.bind(deferred));
     return deferred;
 }
+
+//================================
+// iterators
+
+// tslint:disable-next-line:promise-must-complete no-empty
+const NEVER = new Promise(() => {});
```

Comment:
- What is the difference between instantiating a promise like this and doing 

```suggestion
const NEVER = Promise.resolve();
```
- Is this constant going to be used in places other than `chainRaw`?
- Adding a type of `Promise<unknown>` allows you to remove the type assertions in `chainRaw` as well.

```suggestion
const NEVER: Promise<unknown> = new Promise(() => {});
```

Reply:
I think `new Promise(() => {})` and `Promise.resolve()` are entirely different things. The first promise never resolves meanwhile the latter one returns an already resolved promise

Reply:
Do we need a pending promise here, or can we work with `Promise.resolve()` directly?

Reply:
We're doing a `Promise.race(promises)` in line 168. There we need a promise which never resolves, so no we can't work with `Promise.resolve()`.

Reply:
> Adding a type of Promise<unknown> allows you to remove the type assertions in chainRaw as well.

I think this suggestion was missed, unresolving.